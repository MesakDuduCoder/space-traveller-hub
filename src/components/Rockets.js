import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { cancelReserveRocket, reserveRocket } from '../redux/rockets/rocketSlice';
import '../css/rocket.css';

function Rockets() {
  const dispatch = useDispatch();
  const { rockets, isLoading } = useSelector((store) => store.rockets);

  if (isLoading) {
    return (
      <div>Loading......</div>
    );
  }

  return (
    <div className="rockets">
      {rockets.map((rocket) => (
        <div className="rocket" key={rocket.id}>
          <div className="imageBox">
            <img src={rocket.flickr_images} alt="" className="rocketImage" />
          </div>
          <div className="rocketDetails">
            <h2>{rocket.name}</h2>
            <div>
              {rocket.reserved ? (
                <span className="reserved" data-testid="rocketReserved">
                  Reserved
                </span>
              ) : (
                ''
              )}
              <span>{rocket.description}</span>
            </div>
            {rocket.reserved && (
              <button
                type="button"
                className="cancelRocket"
                onClick={() => dispatch(cancelReserveRocket(rocket.id))}
              >
                Cancel reservation
              </button>
            )}
            {!rocket.reserved && (
              <button
                type="button"
                data-testid="reserveBtn"
                className="reserveRocket"
                onClick={() => dispatch(reserveRocket(rocket.id))}
              >
                Reserve Rocket
              </button>
            )}
          </div>
        </div>
      ))}
    </div>
  );
}

export default Rockets;
