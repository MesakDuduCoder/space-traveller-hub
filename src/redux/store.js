import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketSlice';
import missionReducer from './mission/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
    rockets: rocketsReducer,
  },
});

export default store;
