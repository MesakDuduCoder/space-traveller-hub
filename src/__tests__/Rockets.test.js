import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import React from 'react';
import store from '../redux/store';
import Rockets from '../components/Rockets';

describe('User Interactions test', () => {
  it('will test user interaction', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
