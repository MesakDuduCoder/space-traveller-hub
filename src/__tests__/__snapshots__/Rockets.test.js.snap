// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<div>
  Loading......
</div>
`;
